src/adapter/src/session/vars.rs

```diff
@@ -1174,3 +1182,46 @@ impl Value for TimeZone {
         self.as_str().into()
     }
 }
+
+/// List of valid isolation levels.
+#[derive(Clone, Copy, Debug, Eq, PartialEq)]
+pub enum IsolationLevel {
+    Serializable,
+    StrictSerializable,
+}
+
+impl IsolationLevel {
+    fn as_str(&self) -> &'static str {
+        match self {
+            IsolationLevel::Serializable => "SERIALIZABLE",
+            IsolationLevel::StrictSerializable => "STRICT_SERIALIZABLE",
+        }
+    }
+
+    fn valid_values() -> Vec<&'static str> {
+        vec![
+            IsolationLevel::Serializable.as_str(),
+            IsolationLevel::StrictSerializable.as_str(),
+        ]
+    }
+}
+
+impl Value for IsolationLevel {
+    const TYPE_NAME: &'static str = "string";
+
+    fn parse(s: &str) -> Result<Self::Owned, ()> {
+        let s = UncasedStr::new(s);
+
+        if s == IsolationLevel::Serializable.as_str() {
```

Comment:
Would it make sense to add all the levels below serializable and have this function return the correct level? Then just we treat all levels except for strict serializability the same? I'm just thinking that it might be confusing to a user if they do the following:
```
materialize=> SET transaction_isolation = READ_COMMITTED;
SET

materialize=> SHOW transaction_isolation;
 transaction_isolation 
-----------------------
 SERIALIZABLE
(1 row)
```

Reply:
Ah I thought we already did this, I was wrong. It's fine as is. Cockroach also complains. For some reason I thought I added this because an ORM wanted it but I must be thinking about something else.

Reply:
No I think you are thinking of the right thing, I just found it.

https://github.com/MaterializeInc/materialize/blob/8e54ed5949634a4a0b57283f53c15bb96e74b15c/src/sql-parser/src/ast/defs/statement.rs#L2087-L2093

These get parsed, and then converted into a no-op. I'll have to do some plumbing to connect these to what I added.

Reply:
Just added this in.